packages/core/src/config/config.ts

```diff
@@ -266,12 +269,20 @@ export class Config {
   }
 
   setModel(newModel: string): void {
+    if (this.forceModel) {
```

Comment:
setModel should set the model no matter what but we should audit call sites to ensure we do not set this when forceModel is set unless the user has confirmed.